tests/components/onkyo/test_config_flow.py

```diff
@@ -403,14 +354,15 @@ async def test_reconfigure_new_device(hass: HomeAssistant) -> None:
 
     result = await config_entry.start_reconfigure_flow(hass)
 
-    receiver_info_2 = create_receiver_info(2)
+    mock_connection = create_connection(2)
 
-    with patch(
-        "homeassistant.components.onkyo.config_flow.async_interview",
-        return_value=receiver_info_2,
-    ):
+    # Create mock discover that calls callback immediately
+    async def mock_discover(host, discovery_callback, timeout):
+        await discovery_callback(mock_connection)
+
+    with patch("pyeiscp.Connection.discover", new=mock_discover):
```

Comment:
Can this be patched at the entry point in the integration?

Reply:
Ah my bad forgot to limit patches to this test scope. 

Is this good?
"homeassistant.components.onkyo.receiver.pyeiscp.Connection.discover"

Reply:
Yes